TcUnit-Runner/VisualStudioInstance.cs

```diff
@@ -107,29 +107,56 @@ public void Close()
             loaded = false;
         }
 
-        private void LoadDevelopmentToolsEnvironment(string visualStudioVersion)
+        private void LoadDevelopmentToolsEnvironment(string visualStudioVersion, bool pinnedTCVersion)
         {
             /* Make sure the DTE loads with the same version of Visual Studio as the
              * TwinCAT project was created in
              */
             
             // Load the DTE
             string VisualStudioProgId = VisualStudioDteAvailable(visualStudioVersion);
-
+            bool versionAvailable = false;
             dte.UserControl = false; // have devenv.exe automatically close when launched using automation
             dte.SuppressUI = true;
             // Make sure all types of errors in the error list are collected
             dte.ToolWindows.ErrorList.ShowErrors = true;
             dte.ToolWindows.ErrorList.ShowMessages = true;
             dte.ToolWindows.ErrorList.ShowWarnings = true;
+            // First set the SilentMode and then try to open the Remote Manager
+            var tcAutomationSettings = dte.GetObject("TcAutomationSettings");
+            tcAutomationSettings.SilentMode = false; // Only available from TC3.1.4020.0 and above
 
             // Load the correct version of TwinCAT using the remote manager in the automation interface
             log.Info("Using the TwinCAT remote manager to load TwinCAT version '" + this.tcVersion + "'...");
             ITcRemoteManager remoteManager = dte.GetObject("TcRemoteManager");
-            remoteManager.Version = this.tcVersion;
+            
+            //Check if the Version is Pinned, if the Version is Pinned check if the Pinned Version is available
```

Comment:
Try to use same formatting for comments as rest of the code

Reply:
I try to see it but do not see the error.
I have chosen the same format as in step // Load the correct version.

Please let me know what exactly I should change
`
            // Load the correct version of TwinCAT using the remote manager in the automation interface
            log.Info("Using the TwinCAT remote manager to load TwinCAT version '" + this.tcVersion + "'...");
            ITcRemoteManager remoteManager = dte.GetObject("TcRemoteManager");
            
            // Check if the Version is Pinned, if the Version is Pinned check if the Pinned Version is available
            if (pinnedTCVersion)`

Reply:
Yes, this is fine. Thanks!